integration/tests/managed/managed_nodegroup_test.go

```diff
@@ -142,504 +140,387 @@ var _ = Describe("(Integration) Create Managed Nodegroups", func() {
-			params.LogStacksEventsOnFailure()
+		It("should have functional DNS", func() {
+			d := test.CreateDaemonSetFromFile(test.Namespace, "../../data/test-dns.yaml")
+			test.WaitForDaemonSetReady(d, defaultTimeout)
+			{
+				ds, err := test.GetDaemonSet(test.Namespace, d.Name)
+				Expect(err).ShouldNot(HaveOccurred())
+				fmt.Fprintf(GinkgoWriter, "ds.Status = %#v", ds.Status)
+			}
+		})
 
-			It("should create nodegroups with taints applied", func() {
-				taints := []api.NodeGroupTaint{
-					{
-						Key:    "key1",
-						Value:  "value1",
-						Effect: "NoSchedule",
-					},
-					{
-						Key:    "key2",
-						Effect: "NoSchedule",
-					},
-					{
-						Key:    "key3",
-						Value:  "value2",
-						Effect: "NoExecute",
-					},
-				}
-				clusterConfig := makeClusterConfig()
-				clusterConfig.ManagedNodeGroups = []*api.ManagedNodeGroup{
-					{
-						NodeGroupBase: &api.NodeGroupBase{
-							Name: "taints",
-						},
-						Taints: taints,
-					},
-				}
+		It("should have access to HTTP(S) sites", func() {
+			d := test.CreateDaemonSetFromFile(test.Namespace, "../../data/test-http.yaml")
+			test.WaitForDaemonSetReady(d, defaultTimeout)
+			{
+				ds, err := test.GetDaemonSet(test.Namespace, d.Name)
+				Expect(err).ShouldNot(HaveOccurred())
+				fmt.Fprintf(GinkgoWriter, "ds.Status = %#v", ds.Status)
+			}
+		})
+	})
 
-				cmd := params.EksctlCreateCmd.
-					WithArgs(
-						"nodegroup",
-						"--config-file", "-",
-						"--verbose", "4",
-					).
-					WithoutArg("--region", params.Region).
-					WithStdin(clusterutils.Reader(clusterConfig))
-				Expect(cmd).To(RunSuccessfully())
-
-				config, err := clientcmd.BuildConfigFromFlags("", params.KubeconfigPath)
-				Expect(err).NotTo(HaveOccurred())
-				clientset, err := kubernetes.NewForConfig(config)
-				Expect(err).NotTo(HaveOccurred())
-
-				mapTaints := func(taints []api.NodeGroupTaint) []corev1.Taint {
-					var ret []corev1.Taint
-					for _, t := range taints {
-						ret = append(ret, corev1.Taint{
-							Key:    t.Key,
-							Value:  t.Value,
-							Effect: t.Effect,
-						})
-					}
-					return ret
-				}
+	Context("adding new managed nodegroups", func() {
+		params.LogStacksEventsOnFailure()
+
+		It("supports a public nodegroup", func() {
+			By("creating it")
+			cmd := params.EksctlCreateCmd.WithArgs(
+				"nodegroup",
+				"--cluster", params.ClusterName,
+				"--nodes", "4",
+				"--managed",
+				"--instance-types", "t3a.xlarge",
+				publicNodeGroup,
+			)
+			Expect(cmd).To(RunSuccessfully())
 
-				tests.AssertNodeTaints(tests.ListNodes(clientset, "taints"), mapTaints(taints))
-			})
+			By("ensuring it is healthy")
+			checkNg(publicNodeGroup)
+
+			By("deleting it")
+			cmd = params.EksctlDeleteCmd.WithArgs(
+				"nodegroup",
+				"--verbose", "4",
+				"--cluster", params.ClusterName,
+				publicNodeGroup,
+			)
+			Expect(cmd).To(RunSuccessfully())
+		})
+
+		It("supports a private nodegroup", func() {
+			By("creating it")
```

Comment:
Here and elsewhere, we should make all `By` messages more specific to the test specs.